homeassistant/components/goalzero/const.py

```diff
@@ -21,20 +23,12 @@
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
 
 BINARY_SENSOR_DICT = {
-    "v12PortStatus": ["12V Port Status", CONF_BINARY_SENSORS, None],
-    "usbPortStatus": ["USB Port Status", CONF_BINARY_SENSORS, None],
-    "acPortStatus": ["AC Port Status", CONF_BINARY_SENSORS, None],
-    "backlight": [
-        "Backlight",
-        CONF_BINARY_SENSORS,
-        ["mdi:clock-digital", "mdi:clock-digital"],
-    ],
-    "app_online": [
-        "App Online",
-        CONF_BINARY_SENSORS,
-        ["mdi:lan-disconnect", "mdi:lan-check"],
-    ],
-    "isCharging": ["Is Charging", CONF_BINARY_SENSORS, "mdi:battery-charging"],
+    "v12PortStatus": ["12V Port Status", DEVICE_CLASS_POWER, "mdi:toggle-switch"],
```

Comment:
Drop the icon if you set a device_class as the frontend will take care of it for you.

Reply:
The icons I have set better represent the entities than the ordinary device class ones. Must we remove my icons set? Or should I just remove the device classes for them?

Reply:
If the device class accurately represents the entity according to the docs (eg https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes), don't set an icon.  If the device class does not accurately represent the entity, remove the device class and then set the icon.